Dockerfile

```diff
@@ -28,13 +28,12 @@ RUN apt-get -qq update \
   && rm /tmp/packages-microsoft-prod.deb \
   && curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add - \
   && echo 'deb [arch=amd64] https://download.docker.com/linux/debian buster stable' >> /etc/apt/sources.list \
-  && curl -fsSL https://dl-ssl.google.com/linux/linux_signing_key.pub | gpg --dearmor -o /usr/share/keyrings/dart.gpg \
-  && echo 'deb [signed-by=/usr/share/keyrings/dart.gpg arch=amd64] https://storage.googleapis.com/download.dartlang.org/linux/debian stable main' | tee /etc/apt/sources.list.d/dart_stable.list \
+  && curl -fsSL https://storage.googleapis.com/flutter_infra_release/releases/stable/linux/flutter_linux_3.7.1-stable.tar.xz -o /tmp/flutter_linux_3.7.1-stable.tar.xz \
```

Comment:
We cannot use `snapd`.
I've not found a `/latest` endpoint, so it's the [latest pinned version](https://docs.flutter.dev/get-started/install/linux) as of now.

Reply:
You should not use a dynamic 'latest' version to make image building as idempotent as possible anyway 😉

Reply:
Not sure if this applies to files in tmp as well but ideally you should remove this tar file after extracting.

Reply:
I also prefer pinned versions, this PR is only needed because the new dart version has breaking changes.